cpp/src/partitioning/partitioning.cu

```diff
@@ -728,7 +728,7 @@ std::pair<std::unique_ptr<table>, std::vector<size_type>> hash_partition(
 
   // Return empty result if there are no partitions or nothing to hash
   if (num_partitions <= 0 || input.num_rows() == 0 || table_to_hash.num_columns() == 0) {
-    return std::pair(empty_like(input), std::vector<size_type>{});
+    return std::pair(empty_like(input), std::vector<size_type>(num_partitions, 0));
```

Comment:
why not `num_partitions + 1` here as well?

Reply:
That's also my question for the creator of these APIs. They were designed in that way from the beginning and I don't want to break that.

Reply:
Humnn, just realized that the output like this is wrong: since the output table is empty, all-zeros offsets are invalid. The output should be empty offsets like previously. So the docs should be corrected instead. WIP on this....

Reply:
All zero offsets are not invalid, they indicate every partition is empty (start == end).  I believe the documentation is correct and it is the implementation that is wrong.  The caller specified N partitions should be createds, and returning less than that in some corner case is surprising.

Reply:
Sorry for the late reply, was on vacation and catching up afterwards.  This needs to be num_partitions+1 as @vuule originally commented, yes?  Each partition is a pair in the output, which is why we need an extra value to give the last partition its pair (i.e.: it's like the offsets vector for lists/strings).  The docs also state that it will always return a vector of N+1 indices but this does not do that.

Reply:
Ah sorry, this is for `hash_partition` instead of `partition`, my mistake.  Apparently the former doesn't handle the partition offsets vector the same way as the latter.  Confusing.